### PR TITLE
Post footer implementation

### PR DESCRIPTION
**Pull request by juniuszhou**
Post footer implementation, post footer could be a web url, image or other type of data. Just store a vec<u8> as post footer for each forum user.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1577,18 +1585,25 @@ impl<T: Trait> Module<T> {
...
+        post_footer: Option<Vec<u8>>,
     ) -> dispatch::Result {
         // Ensure user name is valid
         Self::ensure_user_name_is_valid(&name)?;
 
         // Ensure self introduction is valid
         Self::ensure_user_self_introduction_is_valid(&self_introduction)?;
 
+        // Ensure post footer is valid
+        if post_footer.is_some() {
```
Comment (bedeho) [inline src/lib.rs]: `if let Some(ref post_footer) = opt_post_footer` allows you to match and get parameter in one step, which is safer, then you do not need to do `unwrap`.

**Review by bedeho — CHANGES_REQUESTED** [review]
One minor fix.

There is no test case for check that bad post footer is rejected, but I guess you will do this in the later test PR?

**Comment by juniuszhou** [comment]
> There is no test case for check that bad post footer is rejected, but I guess you will do this in the later test PR?
please see test case create_forum_user_post_footer, which include three cases. post footer is ok, too short and too long.

File: src/lib.rs (hunk 1 above)
Comment (juniuszhou) [reply src/lib.rs]: accepted.

**Review by bedeho — APPROVED** [review]

**Comment by bedeho** [comment]
- This only needs rustfmt to pass format check.
- Please also fix the build warnings.

**Comment by juniuszhou** [comment]
> * This only needs rustfmt to pass format check.
Usually, vs code will format code automatically. I will add manual check step before push code in the future.
> * Please also fix the build warnings.
Building warning from I miss one type test case. I found a bug via add the test case.  I will pay more attention to all warnings. It is helpful.

**Pull request merged**